### DISCUSSION TITLE
How to ignore vscode events from firing within a certain time interval?

### DISCUSSION THREAD

**Discussion by Dayveed117** [microsoft/vscode-discussions#44] [Extension Development QnA] [answered]
For example, if I trigger `onDidChangeTextDocument` events consecutively, with an interval of at maximum 1 second, then I would like to avoid this event's associated logic.
However, if 1 second has passed since the latest `onDidChangeTextDocument` has been triggered, I would like for it to proceed with its logic.

**Comment by r3m0t** [comment, marked as answer]
Use a library like `ts-debounce`? https://www.npmjs.com/package/ts-debounce

**Reply by digitarald** [reply]
This is also what is used to debounce/throttle frequent DOM events like keyboard and mouse.

**Reply by Dayveed117** [reply]
Sorry for the late response.
Yes, this is precisely what I was looking for, cheers!
